### PR TITLE
Ck/make shit less broken

### PR DESCRIPTION
**Pull request by c0d3s**
Evidently a few bugs surfaced over our many months of negligence.  This fixes them, but you should probably double check that.  The main bug was fixed by some nil checking, but I tried to make things a bit more idiomatic where possible with a minimal amount of effort. If I ran across something that was stupid and would take less than approximately sixty seconds to update, I made a change.  This mostly involved deleting shit that was unnecessary, getting rid of nonsensical variable names,  making things more concise, etc.  I can definitely spend more time with it if necessary, but tonight I was busy watching the xfiles reboot.

I'll handle the merge conflicts when we're ready.

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -14,57 +16,41 @@ def create
...
   end
 
-  def autocomplete_username
-    term = params[:q].downcase
-    @users = User.where("lower(username) LIKE '%#{term}%'")
-    respond_to do |format|
-      format.json {
-        render json: @users.map(&:username)
-      }
-    end
```
Comment (arnamak) [inline app/controllers/users_controller.rb]: The reason for having two of the same thing here was that with two autocomplete bars (each making their own AJAX requests), it won't work if they share the same method. At least, I could not find another way near the end of that ridiculous 40-hour day we spent on this. 

Did you test it after removing this method?

**Comment by arnamak** [comment]
Overall, it looks fine. I'm unable to test it locally at the moment (environment conflicts I'll sort out later), but the changes thus far seem like a good and much-needed refactor. 

Nice work @c0d3s and thanks for doing it :+1:

**Comment by c0d3s** [comment]
@arnamak I got it working on the profile page for both tabs -- adding users as friends and adding users to flights.  Is there another place it's used that I'm missing?

Regardless, I'll feel better once a few of ya are able to live test it locally.

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (c0d3s) [inline app/controllers/users_controller.rb]: @arnamak I got it working on the profile page for both tabs -- adding users as friends and adding users to flights. Is there another place it's used that I'm missing?

Regardless, I'll feel better once a few of ya are able to live test it locally.

**Comment by c0d3s** [comment]
...also, do we feel okay with the GNU GPL license?

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (arnamak) [inline app/controllers/users_controller.rb]: @c0d3s Ah ok. I think @djvoa12's last minute changes to include tabs might've fixed the problem. At the time I was working on it, it was all on the same page. 

I'm slowly working on being able to test locally, but putting off dealing with weird postgres issues for now. I'll get to it this weekend.

**Comment by arnamak** [comment]
I'm generally more of a fan of the MIT license, but I won't make a fuss about it. Whatever ya'll want is fine by me.

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (daniellekienzle) [inline app/controllers/users_controller.rb]: I'm having issues with postgres and running locally too. I need to spend some time resolving ruby env issues this weekend. :person_frowning:

**Comment by daniellekienzle** [comment]
Everything looks good to me :+1: :100:

**Comment by ryanbjones** [comment]
This brings me joy just seeing this conversation

**Comment by arnamak** [comment]
@ryanbjones :heart:

**Comment by c0d3s** [comment]
@ryanbjones :doughnut: :doughnut: :doughnut: :smile:

**Comment by ryanbjones** [comment]
Man I miss 4 crusty donuts for $1 at 5:50 pm

**Pull request merged**